src/compile/layout/header.ts

```diff
@@ -84,6 +84,28 @@ export function getTitleGroup(model: Model, channel: HeaderChannel) {
   };
 }
 
+export function getLabelAlign(angle: number) {
+  angle %= 360;
+  if (angle > 270 || angle <= 45) {
+    return {align: {value: 'right'}};
+  } else if (angle >= 135 && angle < 225) {
+    return {align: {value: 'left'}};
+  }
+  return {};
+}
+
+export function getLabelBaseline(angle: number) {
+  angle %= 360;
+  if (angle <= 22.5 || angle >= 225 && angle < 270) {
+    return {baseline: {value: 'middle'}};
+  } else if (angle > 22.5 && angle <= 45 || angle > 135 && angle <= 180) {
```

Comment:
- Don't use || and && together without parentheses -- it's very hard to read.  
- Also  do  `1 < a && a < 3` rather than `a > 1 && a < 3`